espnet3/trainer/trainer.py

```diff
@@ -88,19 +86,18 @@ def __init__(
             self._del_config_key("plugins")
 
         # Callbacks
-        # Temporarily disabled for the code review.
-        # callbacks = get_default_callbacks(
-        #     expdir,
-        #     self.config.log_every_n_steps,
-        #     OmegaConf.to_container(best_model_criterion),
-        # )
-        # if getattr(self.config, "callbacks", None):
-        #     assert isinstance(
-        #         self.config.callbacks, ListConfig
-        #     ), "callbacks should be a list"
-        #     for callback in self.config.callbacks:
-        #         callbacks.append(instantiate(callback))
-        #     self._del_config_key("callbacks")
+        callbacks = get_default_callbacks(
+            expdir,
+            self.config.log_every_n_steps,
+            OmegaConf.to_container(best_model_criterion),
+        )
+        if getattr(self.config, "callbacks", None):
+            assert isinstance(
+                self.config.callbacks, ListConfig
+            ), "callbacks should be a list"
+            for callback in self.config.callbacks:
+                callbacks.append(instantiate(callback))
```

Comment:
Can it happen that the defined callbacks in `self.config` are duplicates of the default callbacks? In that case, would it better to  detect such cases to display warnings?

Reply:
Thank you, the current implementation does not account for cases where callbacks defined in `get_default_callbacks` overlap with those specified in the config. For example, `LearningRateMonitor` may be registered twice, which can result in duplicated log outputs.. 

In the current behavior of Lightning this is not a critical issue, so it does not immediately cause an error.